bin/workflows/pycbc_make_faithsim_workflow

```diff
@@ -0,0 +1,218 @@
+
+        node.new_output_file_opt(
+            analysis_time, ".dat", "--output", tags=self.tags + extra_tags
+        )
+
+        return node
+
+class FaithSimPlotExecutable(wf.Executable):
+    """Class for running pycbc_faithsim_plot"""
+
+    current_retention_level = wf.Executable.ALL_TRIGGERS
+
+    def create_node(
+        self,
+        analysis_time,
+        input_file,
+        config,
+        subsec,
+        extra_tags=None,
+    ):
+        if extra_tags is None:
+            extra_tags = []
+        node = wf.Executable.create_node(self)
+
+        node.add_input_opt("--input-file", input_file)
+
+        confs = ConfigParser.ConfigParser()
+        confs.read(config)
+
+        arg, value = get_ini_opts(confs, subsec)
+        for i in range(len(arg)):
+            node.add_opt("--" + arg[i], value[i])
+
+        node.new_output_file_opt(analysis_time, ".png", "--output-plot", tags=self.tags)
+
+        return node
+
+
+parser = argparse.ArgumentParser(description=__doc__)
+parser.add_argument("--version", action="version", version=__version__)
+wf.add_workflow_command_line_group(parser)
+wf.add_workflow_settings_cli(parser)
+args = parser.parse_args()
+
+workflow = wf.Workflow(args)
+
+config_path = os.path.abspath(args.config_files[0])
+
+wf.makedir(args.output_dir)
+os.chdir(args.output_dir)
+
+wf.makedir("match")
+wf.makedir("collect_results")
+wf.makedir("injections")
+wf.makedir("bank")
+wf.makedir("log")
+wf.makedir("plots")
+
+num_banks = workflow.cp.get("splitbank", "num_banks")
+
+injections_exe = CreateInjectionExecutable(
+    workflow.cp,
+    "lalapps_inspinj",
+    ifos=["H1", "L1", "V1"],
+    out_dir="injections",
+    tags=["inj"],
+)
+
+inj_node = injections_exe.create_node(workflow.analysis_time)
+workflow += inj_node
+
+inj = inj_node.output_files[0]
+
+split_exe = wf.PycbcSplitBankXmlExecutable(
+    workflow.cp, "pycbc_splitbank", num_banks=num_banks, out_dir="bank"
+)
+spltbank_node = split_exe.create_node(inj)
+
+workflow += spltbank_node
+
+faithsim_exe = FaithsimExecutable(
+    workflow.cp, "pycbc_faithsim", ifos=["H1", "L1", "V1"], out_dir="match"
+)
+
+collect_exe = CollectResultsExecutable(
+    workflow.cp,
+    "pycbc_faithsim_collect_results",
+    ifos=["H1", "L1", "V1"],
+    out_dir="collect_results",
+)
+
+faithsim_files = wf.FileList([])
+
+for i in range(int(num_banks)):
+    bank = spltbank_node.output_files[i]
+    faithsim_node = faithsim_exe.create_node(
+        workflow.analysis_time,
+        param_file=bank,
+        extra_tags=["match-%i" % i],
+    )
+    workflow += faithsim_node
+    faithsim_files.append(faithsim_node.output_file)
+
+collect_node = collect_exe.create_node(
+    workflow.analysis_time,
+    faithsim_files,
+    spltbank_node.output_files,
+    extra_tags=["result-"],
+)
+workflow += collect_node
+collect_results = collect_node.output_files[0]
+
+subsections = [
+    sec for sec in workflow.cp.sections() if sec.startswith("pycbc_faithsim_plots")
+]
```

Comment:
@spxiwh after your other comment is addressed, what is the best way to iterate over the `pycbc_faithsim_plots` subsections with the different tags?

Reply:
The example in the all-sky workflow is things like this https://github.com/gwastro/pycbc/blob/master/pycbc/workflow/plotting.py#L160

`workflow.cp.get_subsections('plot_sensitivity')` would get the subsections directly. But there was some additional code added here to allow other tags (e.g. ifo tags) to also be used without interference. In this instance we also use PlotExecutable directly and just explicitly add commands for the input/output rather than defining a new subclass. I don't have a preference on that vs explicit classes for each type, but you probably do want to inherit from PlotExecutable.